### DISCUSSION TITLE
Consider GPL v2 instead of v3, to match Praat license?

### DISCUSSION THREAD

**Discussion by lenzo-ka** [YannickJadoul/Parselmouth#114] [General]
The code is under GPL v3, the bindings under BSD, and Praat is under GPL v2 license. Would you consider changing the license to GPL v2, so that it matches the Praat licensing? I say this because of the murkiness of some of the language in v3 around data in services, which makes people skittish. I'd love to use the code and not write another set of bindings to avoid v3.

**Comment by YannickJadoul** [comment]
Hi, @lenzo-ka! Thanks for your interest and message!
In general, I don't have a strong opinion here, and "just" wanted an open source license for this project. Back then, I just had a look at https://choosealicense.com/, then realized Praat is GPL so I would also have to go with GPL, then picked the latest version (which was also mentioned on https://choosealicense.com/). As far as I understood, the pybind11 library being released under BSD should not be an issue and should still allow Parselmouth as a whole to be released under GPL?

So I am willing to consider things, but I would like to know if you have a reference to whatever the potential issue is, or a more concrete example of an issue? I'd want to be a bit more about this, but most things I find are just referring to some hardware-related issues which were changed by GPLv3.

Also, Praat's licensing is somewhat confusing, I now found out. The code itself is GPLv2 (or later) indeed (cfr. https://github.com/praat/praat/blob/master/main/GNU_General_Public_License.txt and the headers in the files), but Praat's documentation says it's licensed under GPLv3: https://www.fon.hum.uva.nl/praat/manual/License.html.
The reason they give is that some of Praat's dependencies are GPLv3: https://www.fon.hum.uva.nl/praat/manual/Acknowledgments.html
As things now stand, Parselmouth just includes these (GSL, GLPK, eSpeak NG) through Praat, so I'm not sure I actually have any further choice, here?

Apart from all this, I'm happy when people want to use Parselmouth, especially in scientific and other non-commercial contexts, and happy to further discuss if there's anything limiting this. But I would like to understand better why, in which case you want to use it but couldn't, and I am not convinced there's actually much choice I have, given Praat's situation?

**Reply by lenzo-ka** [reply]
I think you have it right — since praat has gplv3, it wouldn’t make any practical difference to change the license. I appreciate the thoughtful reply. I don’t think there should be a problem, but the gpl in general is written more as a manifesto than as a legal document, and v3 included different language around data that muddies the waters. I am not a lawyer, however. Thanks again for considering it, and explaining the situation.

**Reply by YannickJadoul** [reply]
Thanks! And sorry for the disappointing news, then. I did now adapt the "License" section in the README to better present what I found out. Feel free to let me know if it's still not clear enough!

At any rate, don't hesitate to contact me when necessary; we can always try to figure something out with licenses in a concrete case (as almost all of Parselmouth's code is still under my personal copyright and not contributed by others), or even just for tips and tricks when you would decide to explore the Praat code base yourself!
